catboost/R-package/R/catboost.R

```diff
@@ -396,6 +396,33 @@ print.catboost.Pool <- function(x, ...) {
 }
 
 
+#' @title Print basic information about model
+#' @description Displays the most general characteristics of a CatBoost model.
+#' @param x The model obtained as the result of training.
+#' @param ... Not used
+#' @return The same model that was passed as input.
+#' @export
+print.catboost.Model <- function(x, ...) {
+    cat(sprintf("CatBoost model (%d trees)\n", x$tree_count))
+    cat(sprintf("Evaluation metric: %s\n", catboost.get_plain_params(x)$eval_metric))
+    cat(sprintf("Fit to %d features\n", NROW(x$feature_importances)))
+    if (is.null.handle(x$handle))
+        cat("(Handle is incomplete)\n")
```

Comment:
What is this information for?

Reply:
It’s in order to know if the model object doesn’t contain the underlying C++ object (happens after de-serializing through R’s own functions), so as to know if it’d benefit from calling the function to deserialize it from the raw bytes instead of having it internally de-serialize every time a function is called on the model.